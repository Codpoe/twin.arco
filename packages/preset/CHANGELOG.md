# @twin.arco/preset

## 0.1.1

### Patch Changes

- c6d7a53: Fix: wrong devpendence version

## 0.1.0

### Minor Changes

- Feature: first release
