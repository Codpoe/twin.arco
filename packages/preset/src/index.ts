import { createPalette, createTheme, defaultKeyFormatter } from './utils';

export * from './utils';

export interface PresetOptions {
  keyFormatter?: typeof defaultKeyFormatter;
  enableOpacity?: boolean;
}

export default function (
  { keyFormatter, enableOpacity }: PresetOptions = {
    keyFormatter: defaultKeyFormatter,
    enableOpacity: true,
  },
) {
  return {
    theme: {
      extend: {
        colors: {
          ...createTheme(keyFormatter),
          ...createPalette(enableOpacity, keyFormatter),
        },
      },
    },
  };
}
