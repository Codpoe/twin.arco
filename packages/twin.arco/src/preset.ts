import preset from '@twin.arco/preset';

export default preset;
