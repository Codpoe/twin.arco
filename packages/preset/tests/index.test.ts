import { Processor } from 'windicss/lib/index';
import preset from '@/index';

describe('Twin.arco preset', () => {
  test('WindiCSS preset', () => {
    const processor = new Processor({
      presets: [preset()],
    });
    const result = processor.interpret(
      'text-red-1 bg-red-1 border-red-1 text-arco-text-1 bg-arco-bg-1 border-arco-border-1',
    );
    expect(result.ignored.length).toBe(0);
    expect(result.styleSheet.build()).toMatchSnapshot();
  });
});
