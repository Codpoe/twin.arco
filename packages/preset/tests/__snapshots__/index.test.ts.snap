// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twin.arco preset WindiCSS preset 1`] = `
".bg-red-1 {
  --tw-bg-opacity: 1;
  background-color: rgba(var(--red-1), var(--tw-bg-opacity, 1));
}
.bg-arco-bg-1 {
  background-color: rgb(var(--color-bg-1));
}
.border-red-1 {
  --tw-border-opacity: 1;
  border-color: rgba(var(--red-1), var(--tw-border-opacity, 1));
}
.border-arco-border-1 {
  border-color: rgb(var(--color-border-1));
}
.text-red-1 {
  --tw-text-opacity: 1;
  color: rgba(var(--red-1), var(--tw-text-opacity, 1));
}
.text-arco-text-1 {
  color: rgb(var(--color-text-1));
}"
`;
